src/data/list/nodup.lean

```diff
@@ -0,0 +1,262 @@
+/-
+Copyright (c) 2018 Mario Carneiro. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mario Carneiro, Kenny Lau
+-/
+import data.list.pairwise
+import data.list.forall2
+
+universes u v
+
+open nat function
+
+variables {α : Type u} {β : Type v}
+
+namespace list
+
+/- no duplicates predicate -/
+
+section nodup
```

Comment:
```suggestion
```

Could you move the definition of the `nodup` predicate here as well?

Reply:
Moved.

Reply:
Moved back, the linters use `no_dup`. This is making me nervous about moving things out of `defs`.

Reply:
One of the linters uses `nodup`. I think it's dangerous to move the definitions out of `defs`, as then they become unavailable to tactics. I don't think I'm going to proceed with any of the suggestions to move things out of `defs` for now.

(I will remove the unnecessary `section`s in the new files.)